research.rst

```diff
@@ -0,0 +1,315 @@
+own repo, with the naming scheme ``tequila-<rolename>``.  Each role
+repo would get a tagged release number when changes land in master.
+
+The pros:
+
+- we would not need to commit to publishing to the Ansible Galaxy
+  site, since roles can also be installed by pointing to a repo
+- particular versions of roles can be pinned
+- roles can be installed using a ``requirements.yml`` file similar to
+  a pip requirements file
+- use of the ``ansible-galaxy`` command would allow installed tequila
+  roles to live side-by-side with installed roles from the community
+- conforming to community standards would allow our roles to be used
+  in contexts outside of the django-project-template by other members
+  of the community, using Ansible's standard tools
+
+The cons:
+
+- ``ansible-galaxy`` is very particular about the directory structure
+  of roles that it can install
+- due to that structure, there is a limit of one role per repo or
+  "package"
+- management of versions of all of these repos
+
+For ease of installation, projects will have to ship with an
+ansible-galaxy requirements file, and an ``ansible.cfg`` file
+specifying a ``roles_path`` within the project's directory structure
+(e.g. ``roles/``).  Example versions of these will need to be included
+in django-project-template.
+
+What is the point of the tequila repo, then?
+
+The tequila repo, then, would be a pip-installable central
+clearinghouse for these roles.  It would also get one overall tagged
+release number tying together the release numbers for all of the
+individual roles.  It could also ship with an installation script
+(``tequila roles``) that would call ``ansible-galaxy`` for all of the
+tequila sub-repo versions relevant for the current tequila release.
+
+People would still be free, however, to install individual tequila
+roles by directly using ``ansible-galaxy``.  We may even wish to do
+this for infrequently used roles that are only relevant for some
+projects.
+
+The remaining question is how to deal with development of the tequila
+roles themselves.  How can developers try them out before release?
+For installing feature branches, it should be sufficient to use the
+branch name as the version in the ``ansible-galaxy`` command.  For
+iterative development, we can edit the project's ``ansible.cfg``
+in-place to include the repo for the role at the head of the
+``roles_path`` variable.
+
+
+Proposal 3: all commands must be ``tequila ...`` commands
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This is how the current version of Tequila works.
+
+The ``tequila`` command sets an environment variable for the roles
+path, pointing to the ``tequila/roles`` directory wherever pip
+installed it.
+
+The pros:
+
+- we don't have to worry about a ``roles/`` directory inside our
+  project directory
+
+The cons:
+
+- no longer easily able to use Ansible commands directly
+- our roles are even less usable to the larger community than under
+  Proposal 1
```

Comment:
No, I think you are right now that I think about it.  If someone is an external user, they can still just ``git clone`` the repo and adjust an ``ansible.cfg`` file to include it in their roles path.